flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -1200,48 +1200,63 @@ public boolean receiveAcknowledgeMessage(
      */
     private void completePendingCheckpoint(PendingCheckpoint pendingCheckpoint)
             throws CheckpointException {
-        final long checkpointId = pendingCheckpoint.getCheckpointId();
+        final long checkpointId = pendingCheckpoint.getCheckpointID();
         final CompletedCheckpoint completedCheckpoint;
         final CompletedCheckpoint lastSubsumed;
+        final CheckpointProperties props = pendingCheckpoint.getProps();
 
         // As a first step to complete the checkpoint, we register its state with the registry
-        Map<OperatorID, OperatorState> operatorStates = pendingCheckpoint.getOperatorStates();
-        SharedStateRegistry sharedStateRegistry = completedCheckpointStore.getSharedStateRegistry();
-        sharedStateRegistry.registerAll(operatorStates.values());
+        // we do not
+        if (!props.isSavepoint()) {
```

Comment:
Do I understand correctly that technically we should check `props.isSynchronous()` here as well but we don't do it because we know that job will be stopped after this checkpoint? is it correct?

Reply:
I'd say other way around. We do not register any savepoints in the CheckpointStore (neither intermediate, nor the final/synchronous), thus we should not register their state into shared state.

However we do want to send `notifyCheckpointComplete` for `synchronous` savepoints, because we end the job afterwards and we should leave the system in a consistent state.